include/zephyr/drivers/loopback_disk.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2023 Embedded Solutions GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_LOOPBACK_DISK_ACCESS_H_
+#define ZEPHYR_INCLUDE_DRIVERS_LOOPBACK_DISK_ACCESS_H_
+
+#include <zephyr/drivers/disk.h>
+#include <zephyr/fs/fs_interface.h>
+
+/**
+ * @brief Context object for an active loopback disk device
+ */
+struct loopback_disk_access {
+	struct disk_info info;
+	const char *file_path;
+	struct fs_file_t file;
+	size_t num_sectors;
+};
+
+/**
+ * @brief Register a loopback disk device
+ *
+ * Registers a new loopback disk deviced backed by a file at the specified path.
+ *
+ * @details
+ * @p ctx must be a pointer to a "struct loopback_disk_access" structure that will remain valid for
+ * as long as the disk access is in use.
+ *
+ * @param ctx Preallocated context structure
+ * @param file_path Path to backing file
+ * @param disk_access_name Name of the created disk access (for disk_access_*() functions)
```

Comment:
For the `file_path` and `disk_access_name` parameters, we should either:
- document the requirement that these parameters should not be stack allocated (as they will be assigned to the `ctx` structure)
- make the `file_path` and `disk_access_name` fields in `ctx` be `char []` values, and use `strncpy` to copy the `file_path` and `disk_access_name` parameters provided by the user into those fields

As is stands right now, I worry a user might write code like the following:

```
struct loopback_disk_access global_ctx;
const char *name = "disk_name";

void my_init_func()
{
    const char *file_path = "my_path";

    loopback_disk_access_register(&global_ctx, file_path, name);
}


void write_func()
{
    ....
    /* Path temporary variable will have been deallocated here .. */
    disk_access_read(name,  data_buf, start_sector, num_sector);
}
```

Reply:
That would still be fine, since `file_path` is a string literal, but I get your point. I went with option 1, since I'm not a big fan of statically-sized string buffers.

I've also added an unregister function that will end the lifetime of these parameters.

Reply:
> That would still be fine, since file_path is a string literal, but I get your point. I went with option 1, since I'm not a big fan of statically-sized string buffers.

This is true- perhaps not the best example. Glad it highlighted the point though

Reply:
I think that stack allocated things are ok as long as it is mentioned that they have to be valid through entire runtime.
This already happens when people create file in main of their code and pass it along (or use global to pass it).
And if you allow path to be passed, probably somebody would like to do the
```
snprintf(path, "/my_loopback/disks/%d.img", i);
```
at some point.

Reply:
> I think that stack allocated things are ok as long as it is mentioned that they have to be valid through entire runtime.

I believe that's what the doc comment says, yes.